core/src/main/java/org/infinispan/stream/impl/TerminalFunctions.java

```diff
@@ -615,7 +619,9 @@ private CountFunction() { }
 
       @Override
       public Long apply(Stream<T> stream) {
-         return stream.count();
+         long count = stream.count();
+         log.tracef("Count value is %d", count);
```

Comment:
even though it's trace level, I am not sure if we should log it...

Reply:
Definitely needs logging. The count is computed out of values in local and remotes, and there's no way to see where each part of the count comes from without this.